### DISCUSSION TITLE
New project structure

### DISCUSSION THREAD

**Discussion by sarafarahabadi** [sovelluskontti/project-semantic-search#58] [General]
Hello Turo,

I managed to change the structure of the project completely. Now we have a folder client representing the React app and the server representing the Flask application inside the Docker. It is just the basic code with a simple name test to appear on localhost and it works. 
![10](https://github.com/user-attachments/assets/2d2640b7-3251-4ef5-b37c-be9f3d7d9799)
![image](https://github.com/user-attachments/assets/595bf375-1853-47f7-bd41-fd1f70676aed)

Now both server and client work and are linked. This was the starting test. I will now add the manticore codes to it and start working on the full-text search to appear. 

The new code is now inside the "master" branch. Since it has so many changes compared to the "main" branch. Initially, I wanted to merge master to main and just save the old code in another branch named "oldCode" but I wanted to get your opinion. 
Should I continue working on the master branch this way or do I have to merge it to main. Cause I have to force merge since the differences are fundamental. 

Br, 
Sara

**Comment by sarafarahabadi** [comment]
Hi again, 

I added the Manticore CSV files and Python scripts to the new structure and it works. I can add the data to the manticore table with the new structure. The next step is to start working on the full-text search on the React app. 

Br, 
Sara

**Reply by sarafarahabadi** [reply]
Ps. The final code now is in the "master" branch.

**Reply by nyluntu** [reply]
I think thats fine that we keep only the working code in the end. The old code can be on other branch but let's remove all unused stuff in the end of the project. It's better if the final result is just one main branch that will contain the final state.

Fyi. try to ignore the node_modules and other package management content if possible. The package.json and package-lock file should be enough to install all dependencies. Then the repository will not bloat too big. https://github.com/sovelluskontti/project-semantic-search/tree/master/project-semantic-search/node_modules

Usually that kind of dependencies are ignored.

**Reply by nyluntu** [reply]
Maybe the master branch root folder could have readme with all steps to set up local environment. The project with multiple components can be hard to set up so it's good to keep working instructions with the code. Readme file is enough for that kind of content for other developers.
